crates/yulgen/src/runtime/functions/math.rs

```diff
@@ -287,6 +304,31 @@ fn _checked_exp_unsigned(size: Integer) -> yul::Statement {
     }
 }
 
+fn adjust_numeric_unsigned(size: Integer) -> yul::Statement {
+    if size.is_signed() {
+        panic!("Expected unsigned integer")
+    }
+    let max_value = get_max(size);
+    let fn_name = names::adjust_numeric_size(&size);
+    function_definition! {
+        function [fn_name](value) -> cleaned {
+            (cleaned := and(value, [max_value]))
+        }
+    }
+}
+
+fn adjust_numeric_signed(size: Integer, base: usize) -> yul::Statement {
+    if !size.is_signed() {
+        panic!("Expected signed integer")
+    }
+    let fn_name = names::adjust_numeric_size(&size);
+    function_definition! {
+        function [fn_name](value) -> cleaned {
+            (cleaned := signextend([literal_expression! { (base) }], value))
```

Comment:
Nice, this solution looks good.

Reply:
Just copied what Solidity is doing. We still need to apply that to a bunch of other places (which I will follow up on after this PR)